File: python/bob/bob.py
# -*- encoding: utf-8 -*-

def hey(what):
    what = what.strip()
    if what.isupper():
        return 'Whoa, chill out!'
    if what.endswith('?'):
        return 'Sure.'
    if not what:
        return 'Fine. Be that way!'

    return 'Whatever.'

File: java/bob/Bob.java
import java.util.regex.Pattern;
import java.util.regex.Matcher;

public class Bob {
    public String hey(String what) {
        what = what.strip();
        if (what.isEmpty()) {
            return "Fine. Be that way!";
        }
        Matcher matcher = Pattern.compile("[A-Za-z]").matcher(what);
        if (what.endsWith("?")) {
            if (matcher.find() && what.equals(what.toUpperCase())) {
                return "Calm down, I know what I'm doing!";
            }
            return "Sure.";
        }
        if (matcher.find() && what.equals(what.toUpperCase())) {
            return "Whoa, chill out!";
        }
        return "Whatever.";
    }
}
